profiles/schema.py

```diff
@@ -1369,6 +1371,48 @@ def mutate_and_get_payload(cls, root, info, **input):
             return ClaimProfileMutation(profile=profile_to_claim)
 
 
+def _raise_exception_on_error(info, results):
+    """Raises exception if there were errors and the client didn't request results
+
+    This function can be removed after the client has been updated to read errors
+    from the results.
+    """
+    errors = [error for result in results for error in result.get("errors", [])]
+    if not errors:
+        return
+
+    # No need to raise exception if the client requested results
+    if [
+        field
+        for field in info.field_asts[0].selection_set.selections
+        if field.name.value == "results"
+    ]:
+        return
+
+    dry_run = any(result["dry_run"] for result in results)
+    if dry_run:
+        raise ConnectedServiceDeletionNotAllowedError("Not allowed")
+
+    raise ConnectedServiceDeletionFailedError("Deletion failed")
+
+
+class TranslatedMessageNode(graphene.ObjectType):
+    lang = graphene.String()
+    text = graphene.String()
+
+
+class ErrorNode(graphene.ObjectType):
```

Comment:
Let's call only types that implement the Relay `Node` interface, `<something>Node`.

Also calling this just `Error` is a bit too generic. Perhaps it could be tightly connected with its current usage, which is the error messages from service connection deletion. `ServiceConnectionDeletionError`?

Reply:
Changed